### PR TITLE
Enable containerd OpenLab CI

### PR DESCRIPTION
**Pull request by Yikun**
This is a demo job to trigger the job when pr is submitted, after this pr is merged, every pr in theopenlab/containerd will trigger a `arm-build-job`.

### CODE REVIEW

File: .zuul/playbooks/containerd-build/run.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+          apt-get install -y golang-1.10-go
+          ln -sf ../lib/go-1.10/bin/go /usr/bin/go
+          go get github.com/containerd/containerd
+          cd ~/go/src/github.com/containerd/containerd/
+          make | tee make.txt
+          make test | tee make_test.txt
+          mkdir -p '{{ ansible_user_dir }}/workspace/test_results'
+          cp make*.txt '{{ ansible_user_dir }}/workspace/test_results'
+          cp -r ./bin '{{ ansible_user_dir }}/workspace/test_results'
+        executable: /bin/bash
```
Comment (kiwik) [inline .zuul/playbooks/containerd-build/run.yaml]: EOF line

File: .zuul/playbooks/containerd-build/run.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  tasks:
+    - name: Build containerd
+      shell:
+        cmd: |
+          set -xe
+          apt-get update
+          apt-get install -y btrfs-tools libseccomp-dev git pkg-config
+          apt-get install -y golang-1.10-go
+          ln -sf ../lib/go-1.10/bin/go /usr/bin/go
+          go get github.com/containerd/containerd
```
Comment (kiwik) [inline .zuul/playbooks/containerd-build/run.yaml]: Don't need to get containerd source code, OpenLab Zuul have checkout containerd code and apply related PR code on it.

File: .zuul/playbooks/containerd-build/run.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+- hosts: all
+  become: yes
+  tasks:
+    - name: Build containerd
+      shell:
+        cmd: |
+          set -xe
+          apt-get update
+          apt-get install -y btrfs-tools libseccomp-dev git pkg-config
+          apt-get install -y golang-1.10-go
```
Comment (kiwik) [inline .zuul/playbooks/containerd-build/run.yaml]: You can use OpenLab ansible role `config-golang` to enable specific golang.

**Review by kiwik — CHANGES_REQUESTED** [review]

**Comment by Yikun** [comment]
The CI passed in See https://github.com/Yikun/containerd/pull/5#issuecomment-482406708

File: .zuul/playbooks/containerd-build/run.yaml (hunk 1 above)
Comment (Yikun) [reply .zuul/playbooks/containerd-build/run.yaml]: done

File: .zuul/playbooks/containerd-build/run.yaml (hunk 2 above)
Comment (Yikun) [reply .zuul/playbooks/containerd-build/run.yaml]: done

File: .zuul/playbooks/containerd-build/run.yaml (hunk 3 above)
Comment (Yikun) [reply .zuul/playbooks/containerd-build/run.yaml]: done

File: .zuul/playbooks/containerd-build/run.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      shell:
+        cmd: |
+          set -xe
+          apt-get update
+          apt-get install -y btrfs-tools libseccomp-dev git pkg-config
+          echo $PATH
+          echo $GOPATH
+          mkdir -p $GOPATH/src/github.com/containerd/
+          ln -s '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}' $GOPATH/src/github.com/containerd/containerd
+          cd $GOPATH/src/github.com/containerd/containerd
```
Comment (kiwik) [inline .zuul/playbooks/containerd-build/run.yaml]: Please add a NOTE for the three lines, only forked golang project have to do this workaround, it's not OpenLab limits.
```
          mkdir -p $GOPATH/src/github.com/containerd/
          ln -s '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}' $GOPATH/src/github.com/containerd/containerd
          cd $GOPATH/src/github.com/containerd/containerd
```

File: .zuul/playbooks/containerd-build/run.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+          make | tee make.txt
+          make test | tee make_test.txt
```
Comment (kiwik) [inline .zuul/playbooks/containerd-build/run.yaml]: `make` and `make test` output will be shown in OpenLab job log file `job-output.txt.gz`, if you still need split logs, just need to redirect output to `{{ ansible_user_dir }}/workspace/logs`, see https://github.com/theopenlab/openlab-zuul-jobs/blob/master/roles/copy-files-on-node/tasks/main.yml#L9

File: .zuul/playbooks/containerd-build/run.yaml (hunk 4 above)
Comment (Yikun) [reply .zuul/playbooks/containerd-build/run.yaml]: done

**Review by Yikun — COMMENTED** [review]
@kiwik I add a TODO in here, and once https://github.com/theopenlab/openlab/issues/238 is addressed, we can remove this TODO.

and this TODO can also do a regress test for log refactor.

File: .zuul/playbooks/containerd-build/run.yaml (hunk 4 above)
Comment (kiwik) [reply .zuul/playbooks/containerd-build/run.yaml]: Thank you.

**Review by kiwik — DISMISSED** [review]
LGTM, let us merging the PR when theopenlab/openlab#238 done

**Comment by Yikun** [comment]
The https://github.com/theopenlab/openlab/issues/238 had been merged. The job has been made more clear:
```
make | tee $LOGS_PATH/make.txt
make test | tee $LOGS_PATH/make_test.txt
cp -r ./bin $RESULTS_PATH
```

**Comment by Yikun** [comment]
^ the test passed.

File: .zuul/playbooks/containerd-build/run.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      arch: arm64
+  tasks:
+    - name: Build containerd
+      shell:
+        cmd: |
+          set -xe
+          apt-get update
+          apt-get install -y btrfs-tools libseccomp-dev git pkg-config
+          echo $PATH
+          echo $GOPATH
```
Comment (kiwik) [inline .zuul/playbooks/containerd-build/run.yaml]: Please remove the two debug line.

**Review by kiwik — CHANGES_REQUESTED** [review]

File: .zuul/playbooks/containerd-build/run.yaml (hunk 6 above)
Comment (Yikun) [reply .zuul/playbooks/containerd-build/run.yaml]: done

**Review by kiwik — APPROVED** [review]

**Pull request merged**